src/components/CodeDemo.tsx

```diff
@@ -5,25 +5,62 @@ import { DEMO_URL, PRO_DEMO_URL } from '@/utils/constants'
 
 export type CodeDemoProps = {
   src?: string
-  path: string
+  path?: string
   isPro?: boolean
   isSmall?: boolean
   isLarge?: boolean
   isScrollable?: boolean
+  caption?: React.ReactNode
+  captionActions?: React.ReactNode
 }
 
-export const CodeDemo = ({ src, path, isPro, isSmall, isLarge, isScrollable }: CodeDemoProps) => {
+export const CodeDemo = ({
+  src,
+  path,
+  isPro,
+  isSmall,
+  isLarge,
+  isScrollable,
+  caption,
+  captionActions,
+}: CodeDemoProps) => {
   const iframeHeightClass = isSmall ? 'h-60' : isLarge ? 'h-[70vh] md:h-124' : 'h-96'
+  const hasCaption = caption || captionActions
 
   const iframeProps = {
-    src: src ? src : (isPro ? PRO_DEMO_URL : DEMO_URL) + path,
+    src: src ? src : (isPro ? PRO_DEMO_URL : DEMO_URL) + (path || ''),
```

Comment:
What's the reason having the path being an empty string? That would just load the root path of the demo page, not any demo?